iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -1273,6 +1274,510 @@ public void testScanTableCaseInsensitive() throws IOException {
+        .newInstance(customerSchemaWithNewFirstNameOnly).add(0L, null).add(1L, null).add(2L, null).add(3L, "James");
+    List<Record> customersWithNewFirstNameOnly = customersWithNewFirstNameOnlyBuilder.build();
+
+    // Run a 'select first_name' from Hive to check if the new first-name column can be queried.
+    rows = shell.executeStatement("SELECT customer_id, first_name FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithNewFirstNameOnly,
+        HiveIcebergTestUtils.valueForRow(customerSchemaWithNewFirstNameOnly, rows), 0);
+
+    // Insert data from Hive with first_name filled and with null first_name value.
+    shell.executeStatement("INSERT INTO default.customers values (4L, 'Magenta', 'Lily'), (5L, 'Purple', NULL)");
+
+    // Check if the newly inserted data is returned correctly by select statements.
+    customersWithNewFirstNameBuilder.add(4L, "Magenta", "Lily").add(5L, "Purple", null);
+    customersWithNewFirstName = customersWithNewFirstNameBuilder.build();
+    rows = shell.executeStatement("SELECT * FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithNewFirstName,
+        HiveIcebergTestUtils.valueForRow(customerSchemaWithNewFirstName, rows), 0);
+
+    customersWithNewFirstNameOnlyBuilder.add(4L, "Lily").add(5L, null);
+    customersWithNewFirstNameOnly = customersWithNewFirstNameOnlyBuilder.build();
+    rows = shell.executeStatement("SELECT customer_id, first_name FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithNewFirstNameOnly,
+        HiveIcebergTestUtils.valueForRow(customerSchemaWithNewFirstNameOnly, rows), 0);
+  }
+
+  @Test
+  public void testRenameColumnInIcebergTable() throws IOException {
+    // Create an Iceberg table with the columns customer_id, first_name and last_name with some initial data.
+    Table icebergTable = testTables.createTable(shell, "customers", HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA,
+        fileFormat, HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS);
+
+    // Rename the last_name column to family_name
+    icebergTable.updateSchema().renameColumn("last_name", "family_name").commit();
+
+    Schema schemaWithFamilyName = new Schema(optional(1, "customer_id", Types.LongType.get()),
+        optional(2, "first_name", Types.StringType.get(), "This is first name"),
+        optional(3, "family_name", Types.StringType.get(), "This is last name"));
+
+    // Run a 'select *' from Hive to check if the same records are returned in the same order as before the rename.
+    List<Object[]> rows = shell.executeStatement("SELECT * FROM default.customers");
+    HiveIcebergTestUtils.validateData(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS,
+        HiveIcebergTestUtils.valueForRow(schemaWithFamilyName, rows), 0);
+
+    Schema shemaWithFamilyNameOnly = new Schema(optional(1, "customer_id", Types.LongType.get()),
+        optional(2, "first_name", Types.StringType.get(), "This is first name"));
```

Comment:
Shouldn't this be `family_name`?

Reply:
Yeah, it is a typo. Thanks for finding it. Fixed.